### PR TITLE
[ADD] sale_loyalty_discount_attribute

### PR DESCRIPTION
**Pull request by diogocsc**

### CODE REVIEW

File: sale_loyalty_discount_attribute/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_reward_values_discount(self, reward, coupon, **kwargs):
```
Comment (dreispt) [inline sale_loyalty_discount_attribute/models/sale_order.py:9]: @diogocsc Can you please add a test case for this?

File: sale_loyalty_discount_attribute/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                # sales list price should not be considered for discount as well
+                # and is counted as many times as the product repeats itself in the lines
+                if reward.limit_discounted_attributes == "attributes":
+                    price_unit += (
+                        so_lines.product_id.list_price * len(so_lines) * discount_value
+                    )
+                discount.update({"price_unit": price_unit})
+            return_res.append(discount)
+
+        return return_res
```
Comment (dreispt) [inline sale_loyalty_discount_attribute/models/sale_order.py]: My feeling is that it should return `res`and the `return_Res`variable is not needed.

Comment (diogocsc) [reply sale_loyalty_discount_attribute/models/sale_order.py]: Replaced in latest commit, using an index to update res.

File: sale_loyalty_discount_attribute/models/sale_order.py (hunk 1 above)
Comment (diogocsc) [reply sale_loyalty_discount_attribute/models/sale_order.py:9]: Added unit tests in later commit

**Review by dreispt — APPROVED** [review]

File: sale_loyalty_discount_attribute/tests/test_reward_values_discount.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    def order_line_dict(self, product, qty=1):
+        return {
+            "name": product.name,
+            "product_id": product.id,
+            "product_uom_qty": qty,
+            "product_uom": product.uom_id.id,
+            "price_unit": 10.0,
+        }
+
+    def setUp(self):
```
Comment (rousseldenis) [inline sale_loyalty_discount_attribute/tests/test_reward_values_discount.py:15]: @diogocsc Could you change this to setUpClass ?

Comment (dreispt) [reply sale_loyalty_discount_attribute/tests/test_reward_values_discount.py:15]: Ping @diogocsc

**Comment by dreispt** [comment]
@diogocsc CI is red, can you check please?

**Comment by dreispt** [comment]
Seems like an odd pre-commit install error. Maybe with your new push the build gets green.

File: sale_loyalty_discount_attribute/models/sale_order.py [hunk 4]
```diff
@@ -26,12 +26,16 @@ def _get_reward_values_discount(self, reward, coupon, **kwargs):
                 reward.limit_discounted_attributes
                 and reward.limit_discounted_attributes != "disabled"
             ):
-                for so_lines in self.order_line.filtered(lambda l: not l.reward_id):
+                attributes_wise_price_unit = 0.0
+                for so_lines in self.order_line.filtered(
+                    lambda l: not l.reward_id
+                    and l.product_id.id in reward.discount_product_ids.ids
```
Comment (dreispt) [inline sale_loyalty_discount_attribute/models/sale_order.py:32]: @Nikul-OSI We can also pick a product Category. Would this work in that case?

Comment (Nikul-OSI) [reply sale_loyalty_discount_attribute/models/sale_order.py:32]: @dreispt No, This for only product which is selected on rewards.

image

**Pull request closed without merging**